src/Microsoft.OpenApi.Readers/Document.cs

```diff
@@ -0,0 +1,88 @@
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="diagnostic"></param>
+        /// <param name="settings"></param>
+        /// <returns></returns>
+        public static OpenApiDocument Load(Stream stream, out OpenApiDiagnostic diagnostic, OpenApiReaderSettings settings = null)
+        {
+            return new OpenApiStreamReader(settings).Read(stream, out diagnostic);
+        }
+
+        /// <summary>
+        /// Loads an OpenApiDocument from a TextReader
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="diagnostic"></param>
+        /// <param name="settings"></param>
+        /// <returns></returns>
+        public static OpenApiDocument Load(TextReader reader, out OpenApiDiagnostic diagnostic, OpenApiReaderSettings settings = null)
+        {
+            return new OpenApiTextReaderReader(settings).Read(reader, out diagnostic);
+        }
+
+        /// <summary>
+        /// Loads an OpenApiDocument from a string input
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="diagnostic"></param>
+        /// <param name="settings"></param>
+        /// <returns></returns>
+        public static OpenApiDocument Load(string input, out OpenApiDiagnostic diagnostic, OpenApiReaderSettings settings = null)
+        {
+            return new OpenApiStringReader(settings).Read(input, out diagnostic);
+        }
+
+        /// <summary>
+        /// Loads an OpenApiDocument from a string input
+        /// </summary>
+        /// <param name="input"></param>
+        /// <param name="diagnostic"></param>
+        /// <param name="settings"></param>
+        /// <returns></returns>
+        public static OpenApiDocument Parse(string input, out OpenApiDiagnostic diagnostic, OpenApiReaderSettings settings = null)
+        {
+            return new OpenApiStringReader(settings).Read(input, out diagnostic);
+        }
+
+        /// <summary>
+        /// Loads an OpenApiDocument asynchronously from a TextReader
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="settings"></param>
+        /// <returns></returns>
+        public static Task<ReadResult> LoadAsync(TextReader reader, OpenApiReaderSettings settings = null, CancellationToken cancellationToken = default)
+        {
+            return new OpenApiTextReaderReader(settings).ReadAsync(reader, cancellationToken);
+        }
+
+        /// <summary>
+        /// Loads an OpenApiDocument from a file stream
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <param name="cancellationToken"></param>
+        /// <param name="settings"></param>
+        /// <returns></returns>
+        public static async Task<ReadResult> LoadAsync(Stream stream, OpenApiReaderSettings settings = null, CancellationToken cancellationToken = default)
```

Comment:
why is there any difference with the other loadAsync in terms of async/await?

Reply:
What is the best practice when an method only calls one other method that is async?  Should the inner method be awaited or should we just return the Task?

Reply:
I've seen arguments both ways, but the "perf recommendation" is to avoid instantiating a new task for no reason and just skip async/await to return the task.